packages/cache/src/internal/requestUtils.ts

```diff
@@ -31,32 +32,48 @@ export function isRetryableStatusCode(statusCode?: number): boolean {
   return retryableStatusCodes.includes(statusCode)
 }
 
+async function sleep(milliseconds: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, milliseconds))
+}
+
 export async function retry<T>(
   name: string,
   method: () => Promise<T>,
   getStatusCode: (arg0: T) => number | undefined,
-  maxAttempts = 2
+  maxAttempts = DefaultRetryAttempts,
+  delay = DefaultRetryDelay,
+  onError: ((arg0: Error) => T | undefined) | undefined = undefined
 ): Promise<T> {
-  let response: T | undefined = undefined
-  let statusCode: number | undefined = undefined
-  let isRetryable = false
   let errorMessage = ''
   let attempt = 1
 
   while (attempt <= maxAttempts) {
+    let response: T | undefined = undefined
+    let statusCode: number | undefined = undefined
+    let isRetryable = false
+
     try {
       response = await method()
+    } catch (error) {
+      if (onError) {
+        response = onError(error)
+      }
+
+      isRetryable = true
+      errorMessage = error.message
```

Comment:
Nit: Should we log this errorMessage here since it gets converted to a generic `Cache service responded with ${statusCode}` below if statusCode is set?